(SIMP-7925) Fix simp_grub tests

jeannegreulich opened the pull request:
-  THe simp_grub tests use a fact uniqueid to help generate a hash but
   it is not in all the system facts.

SIMP-7925 #close

trevor-vaughan commented:
@op-ct I think this may be an issue where we're getting Facter v2 facts and `uniqueid` was introduced as standard in Facter v3 so I'm OK with this change.

We should update everything to use Facter v4 and see if that pulls in all of the missing items from v3 (it should).

jeannegreulich commented:
I actually did change the test and that fix was already pushed.  But I thought maybe someone somewhere else might be using the fact and added it in where it was missing.     I changed the test because I couldn't find uniqueid in facter 4 but found it in facte 2.5.  I thought it might be going away in 4.0.

trevor-vaughan commented:
Ouch! We should double check and make sure it's sticking around and check all of our code for usage if it's not.

trevor-vaughan commented:
Did some digging and this is a fact that was dropped after facter 2 so we should not include it.

It's confusing, but our fact sets are actually for facter 3 but we couldn't test cleanly with that so we just sort of shoved them into where they would work. When facter 4 is standard, all of this should right itself.

Closing since this should not be included in the standard fact sets.

op-ct commented:
@jeannegreulich's right: Facter v3 [removed `uniqueid`](https://github.com/puppetlabs/docs-archive/blob/master/facter/3.0/release_notes.md#break-removed-six-facts), and it's still gone in v4.

More generally:

1. We should never alter the factsets by hand―they will break again when re-generated
2. We shouldn't need to alter the factsets to support individual modules' specific test fixtures
   * Instead, explicitly model the requirement in the module's tests with `let(:facts)`

op-ct commented:
(Jinx.)
